sample_app/app_utils.c

```diff
@@ -0,0 +1,714 @@
+   p_api->slots[slot_nbr].name = name;
+   return 0;
+}
+
+int app_utils_pull_module (app_api_t * p_api, uint16_t slot_nbr)
+{
+   if (slot_nbr >= PNET_MAX_SLOTS)
+   {
+      return -1;
+   }
+
+   p_api->slots[slot_nbr].plugged = false;
+
+   return 0;
+}
+
+app_subslot_t * app_utils_plug_submodule (
+   app_api_t * p_api,
+   uint16_t slot_nbr,
+   uint16_t subslot_nbr,
+   uint32_t submodule_ident,
+   const pnet_data_cfg_t * p_data_cfg,
+   const char * submodule_name,
+   app_utils_cyclic_callback cyclic_callback,
+   void * tag)
+{
+   uint16_t subslot_ix;
+
+   if (slot_nbr >= PNET_MAX_SLOTS || p_api == NULL || p_data_cfg == NULL)
+   {
+      return NULL;
+   }
+
+   for (subslot_ix = 0; subslot_ix < PNET_MAX_SUBSLOTS; subslot_ix++)
+   {
+      if (p_api->slots[slot_nbr].subslots[subslot_ix].used == false)
+      {
+         app_subslot_t * p_subslot =
+            &p_api->slots[slot_nbr].subslots[subslot_ix];
+
+         p_subslot->used = true;
+         p_subslot->plugged = true;
+         p_subslot->slot_nbr = slot_nbr;
+         p_subslot->subslot_nbr = subslot_nbr;
+         p_subslot->submodule_name = submodule_name;
+         p_subslot->submodule_id = submodule_ident;
+         p_subslot->data_cfg = *p_data_cfg;
+         p_subslot->cyclic_callback = cyclic_callback;
+         p_subslot->tag = tag;
+         p_subslot->iocs = 0;
+         return p_subslot;
+      }
+   }
+   return NULL;
+}
+
+int app_utils_pull_submodule (
+   app_api_t * p_api,
+   uint16_t slot_nbr,
+   uint16_t subslot_nbr)
+{
+   app_subslot_t * p_subslot = NULL;
+
+   if (slot_nbr >= PNET_MAX_SUBSLOTS || p_api == NULL)
+   {
+      return -1;
+   }
+
+   p_subslot = app_utils_subslot_get (p_api, slot_nbr, subslot_nbr);
+   if (p_subslot != NULL)
+   {
+      memset (p_subslot, 0, sizeof (app_subslot_t));
+      p_subslot->used = false;
+      return 0;
+   }
+
+   return -1;
+}
+
+/**
+ * Get subslot application information.
+ * @param p_appdata        InOut: Application state.
+ * @param slot_nbr         In:    Slot number.
+ * @param subslot_nbr      In:    Subslot number. Range 0 - 0x9FFF.
+ * @return Reference to application subslot,
+ *         NULL if subslot is not found/plugged.
+ */
+app_subslot_t * app_utils_subslot_get (
+   app_api_t * p_api,
+   uint16_t slot_nbr,
+   uint16_t subslot_nbr)
+{
+   uint16_t subslot_ix;
+
+   if (slot_nbr >= PNET_MAX_SLOTS || p_api == NULL)
+   {
+      return NULL;
+   }
+
+   for (subslot_ix = 0; subslot_ix < PNET_MAX_SUBSLOTS; subslot_ix++)
+   {
+      if (p_api->slots[slot_nbr].subslots[subslot_ix].subslot_nbr == subslot_nbr)
+      {
+         return &p_api->slots[slot_nbr].subslots[subslot_ix];
+      }
+   }
+   return NULL;
+}
+
+bool app_utils_subslot_is_input (const app_subslot_t * p_subslot)
+{
+   if (
+      p_subslot != NULL && (p_subslot->data_cfg.data_dir == PNET_DIR_INPUT ||
+                            p_subslot->data_cfg.data_dir == PNET_DIR_IO))
+   {
+      return true;
+   }
+   else
```

Comment:
Maybe replace 
```
else
{
   return false;
}
```
with
```

return false;
```
(note blank line)
Also for the other functions

Reply:
OK